src/include/duckdb/main/client_context.hpp

```diff
@@ -42,6 +44,10 @@ class ClientContext : public std::enable_shared_from_this<ClientContext> {
 	DUCKDB_API explicit ClientContext(shared_ptr<DatabaseInstance> db);
 	DUCKDB_API ~ClientContext();
 
+	//! Previous Query profilers
+	deque<pair<transaction_t, QueryProfiler>> prev_profilers;
```

Comment:
Could you move this to a separate class (QueryProfilerHistory), together with the prev_profilers_size and SetProfilerHistorySize? Make sure these variables are private and that you only operate on prev_profilers and prev_profilers_size through member functions inside the client context, instead of directly operating on internals.

Reply:
Yes, Sure.